docs/docs/02-guide/06-tokenfactory/01-tokenfactory.md

```diff
@@ -0,0 +1,622 @@
+- Remove `TestMsgDeleteDenom_ValidateBasic()` from `messages_denom_test.go`.
+- Eliminate all references to `MsgDeleteDenom()` in `messages_denom.go`.
+
+After making these changes, commit your updates:
+
+```bash
+git add .
+git commit -m "Remove the delete denom functionality"
+```
+
+This concludes the second chapter, setting a solid foundation for your token factory module. In the next chapter, we'll delve into implementing the application logic that will bring your token factory to life.
+
+## Chapter 3: Implementing Core Functionality in Your Token Factory
+
+Having disabled the deletion of denoms, we now turn our attention to the heart of the token factory module: defining the structure of new denoms and implementing their creation and update logic.
+
+**Proto Definition Updates**
+
+Start by defining the structure of a new token denom in `proto/tokenfactory/tokenfactory/tx.proto`.
+
+For `MsgCreateDenom`:
+
+- Remove `int32 supply = 8;` and adjust the field order so `canChangeMaxSupply` becomes the 8th field.
+
+Resulting `MsgCreateDenom` message:
+
+```proto
+message MsgCreateDenom {
+  string owner = 1;
+  string denom = 2;
+  string description = 3;
+  string ticker = 4;
+  int32 precision = 5;
+  string url = 6;
+  int32 maxSupply = 7;
+  bool canChangeMaxSupply = 8;
+}
+```
+
+For `MsgUpdateDenom`:
+
+- Omit `string ticker = 4;`, `int32 precision = 5;`, and `int32 supply = 8;`, and reorder the remaining fields.
+
+Resulting `MsgUpdateDenom` message:
+
+```proto
+message MsgUpdateDenom {
+  string owner = 1;
+  string denom = 2;
+  string description = 3;
+  string url = 4;
+  int32 maxSupply = 5;
+  bool canChangeMaxSupply = 6;
+}
+```
+
+### Client Logic
+
+In the `x/tokenfactory/client/cli/tx_denom.go` file, update the client application logic.
+
+**For `CmdCreateDenom`:**
+
+- Adjust the number of arguments from 8 to 7, removing references to the supply argument, and update the usage descriptions.
+
+**For `CmdUpdateDenom()`:**
+
+- Reduce the number of arguments to 5, excluding `supply`, `precision`, and `ticker`, and modify the usage descriptions accordingly.
+
+Also, update the tests in `x/tokenfactory/client/cli/tx_denom_test.go` to reflect these changes.
+
+### Types Updates
+
+When creating new denoms, they initially have no supply. The supply is determined only when tokens are minted.
+
+In `x/tokenfactory/types/messages_denom.go`:
+
+- Remove the `supply` parameter from `NewMsgCreateDenom`.
+- Update `NewMsgUpdateDenom` to exclude unchangeable parameters like `ticker`, `precision`, and `supply`.
+
+Implement basic input validation in `x/tokenfactory/types/messages_denom.go`:
+
+- Ensure the ticker length is between 3 and 10 characters.
+```go
+func (msg *MsgCreateDenom) ValidateBasic() error {
+    _, err := sdk.AccAddressFromBech32(msg.Owner)
+    if err != nil {
+        return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid owner address (%s)", err)
+    }
```

Comment:
`ErrInvalidAddress` will be deprecated

```suggestion
    if _, err := sdk.AccAddressFromBech32(m.Authority); err != nil {
        return sdkerrors.Wrap(err, "invalid authority address")
    }    
```

Reply:
Ok, let's update when the new version arrives.